utils/src/test/kotlin/OrtAuthenticatorTest.kt

```diff
@@ -66,18 +66,26 @@ class OrtAuthenticatorTest : WordSpec({
             }
         }
 
-        "prefer machine-specific entries over the default" {
+        "recognize a default entry if it is the only entry" {
+            val authentication = getNetrcAuthentication("""
+                default login foo password bar
+            """.trimIndent(), "gitlab.com")
+
+            authentication shouldNotBeNull {
+                userName shouldBe "foo"
+                password shouldBe "bar".toCharArray()
+            }
+        }
+
+        "disregard a default entry that comes before machine entries" {
             val authentication = getNetrcAuthentication("""
                 default login foo password bar
                 machine github.com
                 login git
                 password hub
-            """.trimIndent(), "github.com")
+            """.trimIndent(), "gitlab.com")
 
-            authentication shouldNotBeNull {
-                userName shouldBe "git"
-                password shouldBe "hub".toCharArray()
-            }
+            authentication should beNull()
```

Comment:
This assumption is wrong.
The `.netrc` contains a default entry. It should be used for `gitlab.com`.

Reply:
But the default entry is the *first* in the file, so it should not be used, or?

Reply:
The specification does not mention what should happen if the default entry is **not** placed last.

But Git's parsing of the `.netrc` file uses the `default` entry if one is available before a machine entry.
https://github.com/git/git/blob/7e391989789db82983665667013a46eabc6fc570/contrib/credential/netrc/git-credential-netrc.perl

This can be checked by using such a .netrc file:
```
default login <valid-user> password <valid-password>
machine github.com login <invalid-user> password <invalid-password>
```
and then try to clone or pull from github.